browser/ui/webui/brave_new_tab_message_handler.h

```diff
@@ -6,22 +6,29 @@
 #ifndef BRAVE_BROWSER_UI_WEBUI_BRAVE_NEW_TAB_MESSAGE_HANDLER_H_
 #define BRAVE_BROWSER_UI_WEBUI_BRAVE_NEW_TAB_MESSAGE_HANDLER_H_
 
+#include "base/values.h"
 #include "components/prefs/pref_change_registrar.h"
+#include "chrome/browser/search/instant_service_observer.h"
+#include "chrome/common/search/instant_types.h"
 #include "content/public/browser/web_ui_message_handler.h"
 
+class InstantService;
 class Profile;
 namespace content {
 class WebUIDataSource;
 }
 
 // Handles messages to and from the New Tab Page javascript
-class BraveNewTabMessageHandler : public content::WebUIMessageHandler {
+class BraveNewTabMessageHandler : public content::WebUIMessageHandler,
+                                  public InstantServiceObserver {
```

Comment:
I think brave_new_tab_message_handler.cc is getting a bit too long, and since all the InstantService related messages are new and seem self-contained, perhaps we should make an `InstantServiceMessageHandler` as a sibling to `BraveNewTabMessageHandler` and `BraveNewTabWebUI` can instantiate both.

Reply:
This is good feedback- breaking out to a separate handler would make it easy to pull in just that functionality too... if we wanted to use it somewhere else (like in a sidebar, etc). Will check this out now 😄

Reply:
Done! 😄👍